plugins/pulp_rpm/plugins/migrations/0020_nested_drpm_directory.py

```diff
@@ -0,0 +1,119 @@
+    """
+    Remove all prestodelta repo_content_units since they should nto have been created
+    to begin with
+    """
+    metadata_collection = get_collection('units_yum_repo_metadata_file')
+    repo_units_collection = get_collection('repo_content_units')
+    for presto_unit in metadata_collection.find({'data_type': 'prestodelta'}):
+        # remove any repo repo units that reference it, the unit itself will
+        # be removed by the orphan cleanup at some point in the future
+        repo_units_collection.remove({'unit_id': presto_unit['_id']})
+
+
+def _remove_prestodelta_from_repomd(repo_dir, prestodelta_name):
+    """
+    For a given prestodelta filename remove references to it from
+    the repomd.xml file for a repository
+
+    :param repo_dir: The root directory of the repository to process
+    :type repo_dir: str
+    :param prestodelta_name: The filename of the prestodelta file
+    :type prestodelta_name: str
+    """
+    repomd = os.path.join(repo_dir, 'repodata', 'repomd.xml')
+    presto_name = os.path.join('repodata', prestodelta_name)
+    if os.path.exists(repomd):
+        tree = le.parse(repomd)
+        xpath_str = ".//*[@href='%s']" % presto_name
+        root = tree.getroot()
+        for bad in tree.findall(xpath_str):
+            bad_parent = bad.getparent()
+            root.remove(bad_parent)
+
+        with open(repomd, 'w') as handle:
+            handle.write(le.tostring(root))
+
+
+def _remove_prestodelta_symlinks(repo_dir):
+    """
+    Remove all the prestodelta symlinks from the repodata directory of the given repo
+    In addition links to the bad prestodelta files will be removed from the repomd.
+    Only symlinked prestodeltas are examined since they are links to content units
+
+    :param repo_dir: The root of the repository
+    :type repo_dir: str
+    """
+    # remove xml entries from the repomd that match the symlinks we removed
+    repodata_dir = os.path.join(repo_dir, 'repodata')
+    for file_name in glob.glob(os.path.join(repodata_dir, '*prestodelta*')):
+        if os.path.islink(file_name):
+            os.unlink(file_name)
+            # remove the links from the repomd.xml
+            base_file_name = os.path.basename(file_name)
+            _remove_prestodelta_from_repomd(repo_dir, base_file_name)
+
+
+def _repo_directories(starting_dir):
+    """
+    Generator to find all of the repository directories under a given starting directory.
+
+    :param starting_dir: The directory to search
+    :type starting_dir: str
+    """
+    for dirName, subdirList, fileList in os.walk(starting_dir):
+        if os.path.basename(dirName) == 'repodata':
+            repomd = os.path.join(dirName, 'repomd.xml')
+            if os.path.exists(repomd) and not os.path.islink(repomd):
+                repodir = os.path.abspath(os.path.join(dirName, os.pardir))
+                yield repodir
+
+
+def migrate(*args, **kwargs):
+    """
+    Fix the drpm directories and broken repomd.xml files
```

Comment:
I'm not a PyCharm user, but I think they get mad when you don't have params and types, even when they are unused.

That said, I think it's really stupid that we've added *args and **kwargs to the migrate() functions, just in case one day we might want to put stuff in there. That wasn't my idea - it was forced upon me!

Anyways, if you want to just drop the *args and **kargs, I'd be cool with that too. In fact, I don't really care if you just leave this as is either ☺